tests/unit/windows/TestUwbAppConfiguration.cxx

```diff
@@ -114,8 +135,316 @@ TEST_CASE("UwbAppConfiguration performs allocation for contained value correctly
+    {
+        using namespace windows::devices;
+
+        constexpr DeviceRole roleExpected{ DeviceRole::Initiator };
+
+        UwbSessionData sessionData;
+        sessionData.sessionId = 1;
+        sessionData.uwbConfiguration = UwbConfiguration::Create().DeviceRole(roleExpected);
+        sessionData.uwbConfigurationAvailable = true;
+
+        auto params = GenerateUwbSetAppConfigParameterDdi(sessionData);
+        const auto& ddiParams = params.DdiParameters();
+
+        REQUIRE(ddiParams.sessionId == sessionData.sessionId);
+        REQUIRE(ddiParams.appConfigParamsCount == 1);
+
+        const auto param = ddiParams.appConfigParams[0];
+        const auto sessionUwbMap = sessionData.uwbConfiguration.GetValueMap();
+
+        std::vector<std::unique_ptr<IUwbAppConfigurationParameter>> parameters;
+
+        for (const auto& [parameterTag, parameterValueVariant] : sessionUwbMap) {
+            const auto ddiType = windows::devices::detail::AppConfigUwbConfigurationTagMap.at(parameterTag);
+            REQUIRE(ddiType == param.paramType);
+
+            // the following doesn't want to compile
+            // parameters.push_back(std::make_unique<UwbAppConfigurationParameter<std::underlying_type_t<ParameterValueT>>>(notstd::to_underlying(parameterValueVariant), parameterTag));
+
+            parameters.push_back(std::make_unique<UwbAppConfigurationParameter<DeviceRole>>(roleExpected, parameterTag));
+
+            UwbSetAppConfigurationParameters testStruct{ parameters, sessionData.sessionId };
+
+            const auto& ddiParams = testStruct.DdiParameters();
+
+            REQUIRE(ddiParams.sessionId == sessionData.sessionId);
+            REQUIRE(ddiParams.appConfigParamsCount == std::size(parameters));
+            REQUIRE(ddiParams.size == parameters[0]->DdiSize() + offsetof(UWB_SET_APP_CONFIG_PARAMS, appConfigParams[0]));
+            REQUIRE(std::memcmp(ddiParams.appConfigParams, parameters[0]->DdiBuffer().data(), parameters[0]->DdiSize()) == 0);
+        }
+    }
+
+    SECTION("mac address works")
+    {
+        using namespace windows::devices;
+
+        constexpr std::array<uint8_t, 2> dstMacAddressExpected{ 0xAAU, 0xBBU };
+
+        UwbSessionData sessionData;
+        sessionData.sessionId = 1;
+        sessionData.uwbConfiguration = UwbConfiguration::Create().SetMacAddressControleeShort(dstMacAddressExpected);
+        sessionData.uwbConfigurationAvailable = true;
+
+        auto params = GenerateUwbSetAppConfigParameterDdi(sessionData);
+        const auto& ddiParams = params.DdiParameters();
+
+        REQUIRE(ddiParams.sessionId == sessionData.sessionId);
+        REQUIRE(ddiParams.appConfigParamsCount == 1);
+
+        const auto param = ddiParams.appConfigParams[0];
+        const auto sessionUwbMap = sessionData.uwbConfiguration.GetValueMap();
+
+        std::vector<std::unique_ptr<IUwbAppConfigurationParameter>> parameters;
+
+        for (const auto& [parameterTag, parameterValueVariant] : sessionUwbMap) {
```

Comment:
Like the `enum` test case above, let's get rid of this loop since it's known there's only 1 value. You can use the value helper accessor function `GetControleeShortMacAddress()` to get the value instead.